src/test/java/io/supertokens/test/Utils.java

```diff
@@ -138,46 +137,26 @@ public static void reset() {
     }
 
-    static void commentConfigValue(String key) throws IOException {
+    private static void replaceConfigValue(String regex, String newStr) throws IOException {
         // we close the storage layer since there might be a change in the db related config.
         StorageLayer.close();
+        Path configPath = new File("../config.yaml").toPath();
+        String originalFileContent = Files.readString(configPath);
+        String modifiedFileContent = originalFileContent.replaceAll(regex, newStr);
+        String normalizedFileContent = modifiedFileContent.replaceAll("\r?\n", newLine); // Normalize line endings
+        Files.writeString(configPath, normalizedFileContent);
+    }
 
-        String oldStr = "\n((#\\s)?)" + key + "(:|((:\\s).+))\n";
-        String newStr = "\n# " + key + ":";
-
-        StringBuilder originalFileContent = new StringBuilder();
-        try (BufferedReader reader = new BufferedReader(new FileReader("../config.yaml"))) {
-            String currentReadingLine = reader.readLine();
-            while (currentReadingLine != null) {
-                originalFileContent.append(currentReadingLine).append(System.lineSeparator());
-                currentReadingLine = reader.readLine();
-            }
-            String modifiedFileContent = originalFileContent.toString().replaceAll(oldStr, newStr);
-            try (BufferedWriter writer = new BufferedWriter(new FileWriter("../config.yaml"))) {
-                writer.write(modifiedFileContent);
-            }
-        }
-
+    public static void commentConfigValue(String key) throws IOException {
+        String find = "\r?\n#?\\s*" + Pattern.quote(key) + ":.*\r?\n";
+        String replace = newLine + "# " + key + ":" + newLine;
+        replaceConfigValue(find, replace);
     }
 
     public static void setValueInConfig(String key, String value) throws IOException {
-        // we close the storage layer since there might be a change in the db related config.
-        StorageLayer.close();
-
-        String oldStr = "\n((#\\s)?)" + key + "(:|((:\\s).+))\n";
-        String newStr = "\n" + key + ": " + value + "\n";
-        StringBuilder originalFileContent = new StringBuilder();
-        try (BufferedReader reader = new BufferedReader(new FileReader("../config.yaml"))) {
-            String currentReadingLine = reader.readLine();
-            while (currentReadingLine != null) {
-                originalFileContent.append(currentReadingLine).append(System.lineSeparator());
-                currentReadingLine = reader.readLine();
-            }
-            String modifiedFileContent = originalFileContent.toString().replaceAll(oldStr, newStr);
-            try (BufferedWriter writer = new BufferedWriter(new FileWriter("../config.yaml"))) {
-                writer.write(modifiedFileContent);
-            }
-        }
+        String find = "\r?\n#?\\s*" + Pattern.quote(key) + ":.*\r?\n";
+        String replace = newLine + key + ": " + value + newLine;
```

Comment:
Do we need to change the regex expression? The older one worked just fine. Did the older one cause issues in windows?

Reply:
Yes, this change is necessary.

```
Old Regex: "\n((#\\s)?)" + key                + "(:|((:\\s).+))\n"
New Regex: "\r?\n#?\\s*" + Pattern.quote(key) + ":.*\r?\n"
```

Issues with the Old Regex:
- It used `\n` instead of `\r?\n`, which did not match Windows line endings (`\r\n`)
- It only matched comments with a single whitespace character after it
- It did not match non-commented keys with leading whitespace
- It was also much harder to read due to unnecessary group nesting and the unnecessary `|` operator
- It did not quote the key, so any regex characters in the key would make it fail

When reviewing this, I actually noticed one issue with the New Regex. It did not match commented lines with leading whitespace. I have committed this change.

Reply:
Ok. So this change would be required in the plugins repos as well. Since they too have a similar function in there.

Reply:
Okay, I see. I was thinking that the plugin repos called `io.supertokens.test.Utils.commentConfigValue()` and `io.supertokens.test.Utils.setValueInConfig()` directly.

I was also thinking about renaming `setValueInConfig()` to `setConfigValue()` to match `commentConfigValue()`. If I rename all references in `supertokens-core`, will any change be needed in other repositories?

Reply:
You should change them in the plugin repos too, just to be consistent.